### PR TITLE
Update reactions.dm so that water isn't consumed needlessly

### PR DESCRIPTION
**Pull request by DustanHache**
## About The Pull Request
The reaction information regarding water vapor condensation was inaccurate due to what appears to be a bit of copypasted code when reactions were changed over a bit for excitability.  Water vapor was being consumed wrongly when below freezing temperature. This also indirectly influences electrolysis by making it more efficient (since there's no longer moles vanishing out of existence constantly at safe electrolyzing temperatures.)
## Why It's Good For The Game
Vanishing gases bad. Me want hydrogen.
## Changelog

:cl:

fix: Water vapor is no longer anomalously lost at below freezing temperatures due to condensation. 

:cl:

### CODE REVIEW

**Comment by DustanHache** [comment]
frigging excitables... Whatever was done in that code is causing it to throw warnings, and since I don't fully understand the references being called, I can't do much about it...

**Comment by Pickle-Coding** [comment]
This will cause the reaction to last indefinitely. I think a better way to do this is to have frozen turf effect only expire their status if the conditions to create it are no longer met, and they only check their conditions when the air isn't asleep.

**Comment by DustanHache** [comment]
That would be the ideal, but I have zero clue on how to go about doing that. Besides, this could have unintended behavior if frozen tiles automatically unfroze themselves without atmosphere. It also doesn’t address the issues I am having with the If argument being a PITA and doing nothing.

**Comment by Pickle-Coding** [comment]
The if check leads to no code getting executed. Just remove the if check.

The mole consumption was added on for a reason though, so I don't think it's a good idea to just remove it.

File: code/modules/atmospherics/gasmixtures/reactions.dm [hunk 1]
```diff
@@ -110,7 +110,8 @@
 	switch(air.temperature)
 		if(-INFINITY to WATER_VAPOR_DEPOSITION_POINT)
 			if(location?.freeze_turf())
```
Comment (Pickle-Coding) [inline code/modules/atmospherics/gasmixtures/reactions.dm:112]:
```suggestion
			location?.freeze_turf()
```

The linter error comes from an empty code block in the if check.

File: code/modules/atmospherics/gasmixtures/reactions.dm [hunk 2]
```diff
@@ -109,8 +109,9 @@
 	var/consumed = 0
 	switch(air.temperature)
 		if(-INFINITY to WATER_VAPOR_DEPOSITION_POINT)
-			if(location?.freeze_turf())
-				consumed = MOLES_GAS_VISIBLE
+			location?.freeze_turf()
+			return REACTING
```
Comment (LemonInTheDark) [inline code/modules/atmospherics/gasmixtures/reactions.dm:113]: this means cold watervapor will always be active, even if nothing is going on.
this is troublesome for not wasting processing time, especially since fire creates water currently. 

what is the motivation behind this change? is it just to make hydrogen production more consistent?

Comment (LemonInTheDark) [reply code/modules/atmospherics/gasmixtures/reactions.dm:113]: We can't make the turf being frozen already return a non reaction without making turfs unfreezing/freezing trigger activity. maybe that's the best solution here?

Comment (DustanHache) [reply code/modules/atmospherics/gasmixtures/reactions.dm:113]: it is also to address a inconsistency with gas behavior vs information. the water condensation reaction explicitly states it does not consume moles below 200kelvin. false information makes me angry.

Comment (Pickle-Coding) [reply code/modules/atmospherics/gasmixtures/reactions.dm:113]: > We can't make the turf being frozen already return a non reaction without making turfs unfreezing/freezing trigger activity. maybe that's the best solution here?

Turfs freezing will also freeze everything inside the turf. We could create a freezing component (or element) or something that listens to things entering the turf and freezes them, then removes itself when it gets too hot or there isn't enough water vapour. We can then just make the reaction not return REACTING if the turf already has the freezing component.

Also it looks like you're using the webeditor. I recommend setting up Visual Studio Code and git.

Comment (DustanHache) [reply code/modules/atmospherics/gasmixtures/reactions.dm:113]: I know how to run the server for testing, and managed to do so before undrafting the second time. i don’t quite know my way around the debug tools, so  the testing I was able to do was limited.

Comment (LemonInTheDark) [reply code/modules/atmospherics/gasmixtures/reactions.dm:113]: My concern isn't lag it's just if it works. I need this changed regardless of lag though. We cannot have reactions that go on forever.

Comment (DustanHache) [reply code/modules/atmospherics/gasmixtures/reactions.dm:113]: if you got any suggestions on how to check for frozen turfs, i’m all ears. I’m not good at locating examples of procs outside of the files I have touched.

**Comment by DustanHache** [comment]
No response from Lemon regarding the topic of complaint, so I am closing this.

File: code/modules/atmospherics/gasmixtures/reactions.dm (hunk 2 above)
Comment (LemonInTheDark) [reply code/modules/atmospherics/gasmixtures/reactions.dm:113]: it would be done using a trait on the turf. I think what pickle is suggesting is maybe a bit outside the scope of a first pr, but if you are interested and want help feel free to bug me on discord, I'm @thefinalpotato
